apm-sniffer/apm-sdk-plugin/servicecomb-plugin/servicecomb-java-chassis-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/servicecomb/v2/ProducerOperationHandlerInterceptor.java

```diff
@@ -40,9 +39,14 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         Invocation invocation = (Invocation) allArguments[0];
         ContextCarrier contextCarrier = new ContextCarrier();
         CarrierItem next = contextCarrier.items();
+        boolean inContext = isInContext(invocation);
         while (next.hasNext()) {
             next = next.next();
-            next.setHeadValue(invocation.getContext().get(next.getHeadKey()));
+            if (inContext) {
+                next.setHeadValue(invocation.getContext().get(next.getHeadKey()));
+            }else {
+                invocation.getRequestEx().getHeader(next.getHeadKey());
+            }
```

Comment:
If you want to fail back, don't try to on your own, try to get from context, if empty or null, fall back to from request ext

But still, your codes don't show logic about why this is putting in two places, because client side code doesn't show difference.

Reply:
Client side was determined by the developer. if use cseRestemplate  invocation context and http header contains trace message, but in other methods are only in http header.

Reply:
My question is, where is `cseRestemplate`? I can't see the plugin for it. How it works?

Reply:
no CseRestTemplate plugin. 
Here are two invoke process:
1、RestTemplateBuilder.create().postForObject("http://127.0.0.1:8080/springmvchello/hi", null, String.class); // trace break
consumer --> **RestTemplate** --> provider --> ProducerOperationHandler

2、RestTemplateBuilder.create().postForObject("cse://test/springmvchello/hello", person, String.class);   // trace ok
consumer --> **CseRestTemplate --> TransportClientHandler** --> provider --> ProducerOperationHandler